File: codes/cpp/Code_3_03_1.cpp
#include <iostream>
using namespace std;

int N, A[109];
int Answer = 0;

int main() {
	// 入力
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> A[i];

	// 5 つのカードの番号 (i, j, k, l, m) を全探索
	for (int i = 1; i <= N; i++) {
		for (int j = i + 1; j <= N; j++) {
			for (int k = j + 1; k <= N; k++) {
				for (int l = k + 1; l <= N; l++) {
					for (int m = l + 1; m <= N; m++) {
						if (A[i] + A[j] + A[k] + A[l] + A[m] == 1000) Answer += 1;
					}
				}
			}
		}
	}

	// 答えの出力
	cout << Answer << endl;
	return 0;
}

File: codes/python/Code_3_03_1.py
# 注意
# Python で提出すると、N = 100 では実行に 10 秒程度かかり、TLE（実行時間制限オーバー）になります。
# 一方、同じプログラムを PyPy3 で提出すると、実行に 0.5 秒程度しかかからず、AC（正解）することができます。

# 入力
N = int(input())
A = list(map(int, input().split()))

# 5 つのカードの番号 (i, j, k, l, m) を全探索
answer = 0
for i in range(0, N):
	for j in range(i + 1, N):
		for k in range(j + 1, N):
			for l in range(k + 1, N):
				for m in range(l + 1, N):
					if A[i] + A[j] + A[k] + A[l] + A[m] == 1000:
						answer += 1

# 出力
print(answer)
